server/metrics/index.js

```diff
@@ -0,0 +1,70 @@
+const Prometheus = require('prom-client')
+const { buildPath } = require('../lib/util/logger')
+
+Prometheus.collectDefaultMetrics()
+
+const resolverTimingMetric = new Prometheus.Gauge({
+  name: 'resolver_timing_ms',
+  help: 'Resolver response time in milliseconds',
+  labelNames: ['datasource_type', 'operation_type', 'name']
+})
+
+const resolverRequestsMetric = new Prometheus.Counter({
+  name: 'requests_resolved',
+  help: 'Number of requests resolved by server',
+  labelNames: ['datasource_type', 'operation_type', 'path']
+})
+
+const serverResponseMetric = new Prometheus.Gauge({
+  name: 'server_response_ms',
+  help: 'Server response time in milliseconds',
+  labelNames: ['request_type', 'error']
+})
+
+exports.getMetrics = (req, res) => {
+  res.set('Content-Type', Prometheus.register.contentType)
+  res.end(Prometheus.register.metrics())
+}
+
+exports.responseLoggingMetric = (req, res, next) => {
+  const requestMethod = req.method
+
+  res['requestStart'] = Date.now()
+
+  res.on('finish', onResFinished)
+  res.on('error', onResFinished)
+
+  if (next) next()
+
+  function onResFinished (err) {
```

Comment:
@darahayes @aliok maybe you've got some idea how to avoid nesting of function here?
The reason why I put it here was that I need to log `requestMethod` which wouldn't be a part of the closure if the function was declared outside of this one. Couldn't find an elegant solution.

Reply:
I'm 👍 with this approach